### PR TITLE
Added post function - our use case it to upload the logs to a bucket

### PR DESCRIPTION
**Pull request by daniel-lavie-seekret**

### CODE REVIEW

File: lumberjack.go [hunk 1]
```diff
@@ -38,9 +38,13 @@ import (
 const (
 	backupTimeFormat = "2006-01-02T15-04-05.000"
 	compressSuffix   = ".gz"
-	defaultMaxSize   = 100
+	defaultMaxSize   = 1024 // 1KB
```
Comment (guyarb) [inline lumberjack.go:41]: that's too low
why isn't it configurable?

File: lumberjack.go [hunk 2]
```diff
@@ -82,15 +86,12 @@ type Logger struct {
 	// os.TempDir() if empty.
 	Filename string `json:"filename" yaml:"filename"`
 
-	// MaxSize is the maximum size in megabytes of the log file before it gets
-	// rotated. It defaults to 100 megabytes.
+	// MaxSize is the maximum size in bytes of the log file before it gets
+	// rotated. It defaults to 1 megabytes.
```
Comment (guyarb) [inline lumberjack.go:90]: that's incorrect

File: lumberjack.go [hunk 3]
```diff
@@ -357,9 +361,15 @@ func (l *Logger) millRunOnce() error {
 	}
 
 	for _, f := range remove {
-		errRemove := os.Remove(filepath.Join(l.dir(), f.Name()))
-		if err == nil && errRemove != nil {
-			err = errRemove
+		shouldDelete := true
+		if l.PostOperation != nil {
+			shouldDelete = l.PostOperation.Operate(&f)
```
Comment (guyarb) [inline lumberjack.go:366]: why does "Operate" returns whether we should delete the file or not?

File: lumberjack.go [hunk 4]
```diff
@@ -444,9 +454,9 @@ func (l *Logger) timeFromName(filename, prefix, ext string) (time.Time, error) {
 // max returns the maximum size in bytes of log files before rolling.
 func (l *Logger) max() int64 {
 	if l.MaxSize == 0 {
-		return int64(defaultMaxSize * megabyte)
+		return int64(defaultMaxSize)
```
Comment (guyarb) [inline lumberjack.go:457]: imho you can avoid from the change in the default max size, just override it

**Review by guyarb — CHANGES_REQUESTED** [review]

File: lumberjack.go (hunk 1 above)
Comment (daniel-lavie-seekret) [reply lumberjack.go:41]: That is just default, it is configurable in the sniffer itself

File: lumberjack.go (hunk 2 above)
Comment (daniel-lavie-seekret) [reply lumberjack.go:90]: Fixed

File: lumberjack.go (hunk 3 above)
Comment (daniel-lavie-seekret) [reply lumberjack.go:366]: That gives the user the option to choose if he wants to delete the file. For example- If we could not upload the log to the bucket, we may want to keep it and try next time

File: lumberjack.go (hunk 4 above)
Comment (daniel-lavie-seekret) [reply lumberjack.go:457]: Changed

**Pull request closed without merging**